cmd/ui/src/hooks/useCustomNodeKinds.tsx

```diff
@@ -0,0 +1,9 @@
+import { apiClient } from 'bh-shared-ui';
+import { GetCustomNodeKindsResponse, RequestOptions } from 'js-client-library';
+import { UseQueryResult, useQuery } from 'react-query';
+
+export const getCustomNodeKinds = async (options: RequestOptions): Promise<GetCustomNodeKindsResponse> =>
+    apiClient.getCustomNodeKinds(options).then((res) => res.data);
+
+export const useGetCustomNodeKinds = (): UseQueryResult<GetCustomNodeKindsResponse> =>
+    useQuery<GetCustomNodeKindsResponse>(['getCustomNodeKinds'], ({ signal }) => getCustomNodeKinds({ signal }));
```

Comment:
Great call pulling this out into its own hook using react-query. We use these all the time and are very helpful. I have 2 suggestion:
- I think you can drop the Get from useGetCustomNodeKinds
- Plumb thru the react-query query options so that this hook is more flexible:

```suggestion
export const useGetCustomNodeKinds = (queryOptions?: GenericQueryOptions<GetCustomNodeKindsResponse>): UseQueryResult<GetCustomNodeKindsResponse> => {
    return useQuery({
        queryKey: ['getCustomNodeKinds'],
        queryFn: ({ signal }) => getCustomNodeKinds({ signal }),
        ...queryOptions
    })
}
```